client/src/featureform/register.py

```diff
@@ -601,6 +601,9 @@ def __init__(self, registrar, source):
     def id(self) -> NameVariant:
         return self.__source.name, self.__source.variant
 
+    def name_variant(self) -> NameVariant:
```

Comment:
The fact that it's possible for us to forget this function for some registrars means there should be some contract we need to conform to. Which means an abstract class or interface to implement so that we have consistency. No action item for this PR but something we should do when we refactor.

Reply:
Any other places where we need a name_variant fun?

Reply:
nope, only spot.